dashboard/app/models/concerns/pd/jot_form_backed_form.rb

```diff
@@ -343,13 +331,39 @@ def duplicate?
       new_record? && self.class.exists?(slice(*self.class.unique_attributes_with_form_id))
     end
 
+    # Given answers and questions, returns the correct form of answers.
+    # In the case of a submission where we have the questions, and the survey
+    # model returns true for use_names_for_question_ids?, then the answers have
+    # keys as question names rather than question ID integers where possible.
+    # @param source_answers [Hash] The submitted answers with keys as ID integers.
+    # @param questions [Hash] The questions for this survey.
+    def process_answers_from_submission(source_answers, questions)
+      dest_answers = {}
+
+      source_answers.each do |key, value|
+        # We have an answer ID, but can we get a name instead of the ID?
+        question_detail = questions&.find {|q| q["id"] == key}
+
+        # If we want to use names instead of IDs, and we got a name, then use it.
+        if self.class.use_names_for_question_ids? && question_detail && question_name = question_detail["name"]
+          dest_answers[question_name] = value
+        else
+          dest_answers[key] = value
+        end
+      end
+
+      dest_answers
+    end
+
     # Update answers for this submission from the JotForm API
     # Useful for filling in placeholder response entries (submission id but no answers)
     def sync_from_jotform
       raise 'Missing submission id' unless submission_id.present?
 
       submission = JotForm::Translation.new(form_id).get_submission(submission_id)
-      update!(answers: submission[:answers].to_json)
+      questions_details = self.class.use_names_for_question_ids? ? JSON.parse(questions.questions) : nil
```

Comment:
this means question_details only available if use_names_for_questions_ids is true? Does that affect syncing other forms?

Reply:
The `questions` parameter in `process_answers_from_submission` is optional, and again only attempted to be used when `use_names_for_question_ids?` is `true`.